pkg/dockerclient/shell_test.go

```diff
@@ -0,0 +1,90 @@
+package dockerclient
+
+import (
+	"testing"
+
+	"github.com/nuclio/nuclio/pkg/cmdrunner"
+	"github.com/nuclio/nuclio/pkg/zap"
+
+	"github.com/nuclio/nuclio-sdk"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+)
+
+type MockCmdRunner struct {
+	mock.Mock
+	ExpectedStdOut    string
+	ExpectedStdErr    string
+	ExpectedErrorCode int
+}
+
+func NewMockCmdRunner(expectedStdOut, expectedStdErr string, expectedErrorCode int) *MockCmdRunner {
+	return &MockCmdRunner{
+		ExpectedStdOut:    expectedStdOut,
+		ExpectedStdErr:    expectedStdErr,
+		ExpectedErrorCode: expectedErrorCode,
+	}
+}
+
+func (mcr *MockCmdRunner) Run(options *cmdrunner.RunOptions, format string, vars ...interface{}) (cmdrunner.RunResult, error) {
+	return cmdrunner.RunResult{mcr.ExpectedStdOut, mcr.ExpectedStdErr, mcr.ExpectedErrorCode}, nil
+}
+
+func (mcr *MockCmdRunner) SetStdOut(s string) {
+	mcr.ExpectedStdOut = s
+}
+
+type CmdClientTestSuite struct {
+	suite.Suite
+	logger      nuclio.Logger
+	shellClient ShellClient
+}
+
+func (suite *CmdClientTestSuite) SetupTest() {
+	suite.logger, _ = nucliozap.NewNuclioZapTest("test")
+	tempShellClient, err := NewShellClient(suite.logger, NewMockCmdRunner("", "", 0))
```

Comment:
Why name this `temp` instead of take advantage of the scoping and just use `shellClient`?